aea/cli/utils/package_utils.py

```diff
@@ -0,0 +1,377 @@
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Module with package utils of the aea cli."""
+
+import os
+import re
+import shutil
+from pathlib import Path
+from typing import Dict, Optional
+
+import click
+
+from jsonschema import ValidationError
+
+import yaml
+
+from aea import AEA_DIR
+from aea.cli.utils.constants import NOT_PERMITTED_AUTHORS
+from aea.cli.utils.context import Context
+from aea.cli.utils.loggers import logger
+from aea.configurations.base import (
+    AgentConfig,
+    DEFAULT_AEA_CONFIG_FILE,
+    PackageType,
+    PublicId,
+    _get_default_configuration_file_name_from_type,
+)
+from aea.configurations.loader import ConfigLoader
+from aea.crypto.helpers import (
+    IDENTIFIER_TO_KEY_FILES,
+    _try_validate_private_key_path,
+    create_private_key,
+)
+from aea.crypto.registry import registry
+
+
+def _verify_or_create_private_keys(ctx: Context) -> None:
+    """
+    Verify or create private keys.
+
+    :param ctx: Context
+    """
+    path = Path(DEFAULT_AEA_CONFIG_FILE)
+    agent_loader = ConfigLoader("aea-config_schema.json", AgentConfig)
+    fp = path.open(mode="r", encoding="utf-8")
+    aea_conf = agent_loader.load(fp)
+
+    for identifier, _value in aea_conf.private_key_paths.read_all():
+        if identifier not in registry.supported_crypto_ids:
+            ValueError("Unsupported identifier in private key paths.")
+
+    for identifier, private_key_path in IDENTIFIER_TO_KEY_FILES.items():
+        config_private_key_path = aea_conf.private_key_paths.read(identifier)
+        if config_private_key_path is None:
+            create_private_key(identifier)
+            aea_conf.private_key_paths.update(identifier, private_key_path)
+        else:
+            try:
+                _try_validate_private_key_path(identifier, private_key_path)
+            except FileNotFoundError:  # pragma: no cover
+                raise click.ClickException(
+                    "File {} for private key {} not found.".format(
+                        repr(private_key_path), identifier,
+                    )
+                )
+
+    # update aea config
+    path = Path(DEFAULT_AEA_CONFIG_FILE)
+    fp = path.open(mode="w", encoding="utf-8")
+    agent_loader.dump(aea_conf, fp)
+    ctx.agent_config = aea_conf
+
+
+def _validate_package_name(package_name: str):
```

Comment:
Are any of these functions called outside `packages`? In that case, even if they are standalone functions it may make more sense to include them in that folder, because they are not shared utilities per se.

Reply:
Yes! That's the next goal - to leave private only really private methods. ToDo in next PR cause it will also require many lines changes.

Reply:
Just for clarity, what I mean here is that these methods could be moved to `packages` instead of being in `utils`, if they are only used from `packages`.

Reply:
No. These methods purpose not to be a part of packages. These are the helpers methods for CLI to work with packages - copy, paste, move, etc.